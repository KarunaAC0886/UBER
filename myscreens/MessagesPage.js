import { View, Text, StatusBar,Image } from 'react-native'
import React from 'react'
import BackNavigator from './BackNavigator';
import styles from './styles';

const MessagesPage = () => {
  return (
    <View style={styles.Messages_main_cont}>
    <BackNavigator titleName="Messages" navpath="account"/>
      <View style={styles.Messages_content}>
      <View style={styles.Messages_frst_cont}>
      <Image style={styles.Messages_frst_cont_img} source={require('../assets/Group.png')}/>
      </View>
      <View style={styles.Messages_second_cont}>
      <Text style={styles.Messages_content_frsttext}>You are all up to date!</Text>
      <Text style={styles.Messages_content_scndtext}>No new messages available at the moment, come back soon to discover new offers</Text>
      </View>
      </View>
      <StatusBar/>
    </View>
  )
}

export default MessagesPage