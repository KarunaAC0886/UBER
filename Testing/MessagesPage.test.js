import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import MessagesPage from '../myscreens/MessagesPage';

jest.mock('@expo/vector-icons', () => ({
    FontAwesome5: 'FontAwesome5', 
}));

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

jest.mock('react-native-responsive-screen', () => ({
  widthPercentageToDP: jest.fn(),
  heightPercentageToDP: jest.fn(),
}));
describe('MessagesPage', () => {
  it('Messages Page renders correctly', () => {
    useNavigation.mockReturnValue({ navigate: jest.fn() });
    const tree = renderer.create(<MessagesPage/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});