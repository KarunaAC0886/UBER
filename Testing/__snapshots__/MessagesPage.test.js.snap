// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesPage Messages Page renders correctly 1`] = `
<View
  style={
    {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#000000",
        "height": "13%",
        "justifyContent": "space-between",
        "paddingBottom": "5%",
        "paddingHorizontal": "6%",
        "paddingVertical": "2%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <FontAwesome5
        name="arrow-left"
        style={
          {
            "color": "#ffffff",
            "fontSize": 20,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "Roboto",
          "fontSize": 36,
        }
      }
    >
      Messages
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#1A1A1A",
        "flex": 0,
        "flexDirection": "column",
        "height": "90%",
        "paddingHorizontal": "6%",
        "paddingTop": "15%",
        "paddingVertical": "6%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "40%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/Group.png",
          }
        }
        style={
          {
            "height": 200,
            "width": 290,
          }
        }
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 0,
          "flexDirection": "column",
          "height": "40%",
          "justifyContent": "flex-start",
          "paddingTop": "10%",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 20,
          }
        }
      >
        You are all up to date!
      </Text>
      <Text
        style={
          {
            "color": "#ffffff",
            "fontSize": 18,
            "paddingTop": "4%",
            "textAlign": "center",
          }
        }
      >
        No new messages available at the moment, come back soon to discover new offers
      </Text>
    </View>
  </View>
</View>
`;
